### DISCUSSION TITLE
Confusion over the Install Step

### DISCUSSION THREAD

**Discussion by amirmasoudabdol** [cpm-cmake/CPM.cmake#334] [Q&A]
I'm using CPM to add a few dependencies to my project, and one of them is Boost. So far, I was dealing with building the project, and it all works fine. However, now I want to start shipping it, and prepare it such that CMake can install it too. 

When it comes to Boost, I'm building it via the CPM, and strictly use it as static library by 

```cmake
set(Boost_USE_STATIC_LIBS ON)
find_package(
  Boost REQUIRED)
```

After this, I know that Boost is being used as static library because I don't see traces of it in `otool -L`. However, when I use my `make install`. I end up getting this error which baffles me,

```cmake
Install the project...
-- Install configuration: ""
CMake Error at cmake_install.cmake:42 (include):
  include could not find requested file:

    build/_deps/boost-build/cmake_install.cmake
```

I'm not so fluent with installation steps yet, so this could be some silly mistake in my part but I'm not sure. To me, the installation of CMake should not be triggered at all since I'm using it as static library.

For the reference, I'm adding Boost like below. Btw, I'm not so sure if I'm doing this, `  "CMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}"` correctly. I did it because I wanted to propagate my `BUILD_TYPE` to my dependencies as well. I don't expect this to be a problem since it happens even without it.

```cmake
cpmaddpackage(
  NAME
  Boost
  VERSION
  1.78.0
  OPTIONS
  "BUILD_TESTING:BOOL=OFF"
  "CMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}"    
  "BOOST_INCLUDE_LIBRARIES:STRING=nowide\\\\;filesystem\\\\;system\\\\;date_time\\\\;timer\\\\;chrono\\\\;atomic"
  GITHUB_REPOSITORY
  "boostorg/boost"
  GIT_TAG
  "boost-1.78.0")
```

Also, for some reason, I cannot get the `CPM_USE_LOCAL_PACKAGES` to work. I set it just before adding a package, but it doesn't work!

**Comment by ClausKlein** [comment]
Why not simple do
```cmake
set(BOOST_INCLUDE_LIBRARIES nowide filesystemsystem date_time timer chrono atomic)
CPMAddPackage(...)
```
